### ISSUE TITLE
UMS 10.0.0 - Problem with TV Show Episode File Names.

### ISSUE THREAD

**Issue by Rebel154** [UniversalMediaServer/UniversalMediaServer#2346]
Hi,

I was using V9.8.3 successfully to stream my TV series and movies to my TV and when V10 was released, I updated my installation.

Afterwards, I found that I could not see my TV Series Episodes on my TV's media player (Panasonic 55HZ1500B). I could go into the individual series and seasons, but no episode titles/files appeared. It was the same with the Web Interface - no episode files.

After much pouring over the logs, I found the following line in the full DEBUG Log:

DEBUG 2021-01-15 16:18:34.302 [Library Scanner] Error adding child: "S02E04 - Broken.mkv" java.lang.NullPointerException: Cannot invoke "java.lang.CharSequence.length()" because "this.text" is null

This was the same for every TV episode. The folder/file naming format is 'TV Series/Season 2/S02E01 - Episode Title.mkv'.

I reinstalled V9.8.3 and everything was working again.

So, with V10 running, after some trial and error moving files around the folder structure, I tried renaming the episode files by removing the 'S02' portion, leaving the episode filename as 'E01 - Episode Title.mkv. This worked, and I was able to see the files on my TV and via the web interface.

So in short it appears that UMS 10.0.0 doesn't like the 'S02' part of the filename.

Update 1: I've just tried naming the file 'T02E01 - Episode Title' and that works OK. It seems that it is only 'S' followed by a number as the first 2 characters of the filename cause issues.

I have attached the log files for both 9.8.3 and 10.0.0 as per the Problem Reporting Guidelines

[ums_10.0.0-dbg_2021-01-15-16-21.zip](https://github.com/UniversalMediaServer/UniversalMediaServer/files/5822559/ums_10.0.0-dbg_2021-01-15-16-21.zip)

[ums_dbg_9.8.3-2021-01-15-16-28.zip](https://github.com/UniversalMediaServer/UniversalMediaServer/files/5822560/ums_dbg_9.8.3-2021-01-15-16-28.zip)

Update 2: Some further investigation reveals that the setting of 'Prettify filenames' appears to be causing the issue.

With the setting disabled (unticked), and a restart of the whole program (not just the server) and the episode files show up in the web interface.

Enable the setting (box ticked), and the problem returns following a complete restart of the program.

I've checked both debug logs and the setting was enabled in both versions when I was testing.

**Comment by SubJunk** [comment]
Thanks for reporting this, I will look into this as a priority

**Comment by SubJunk** [comment]
@Rebel154 unrelated, that looks like a custom renderer file, does it work well for you? If so I will add it to the next release

**Comment by SubJunk** [comment]
@Rebel154 I think I have fixed the bug you described but I suggest that if you're naming these files yourself that you use a more standard naming convention like Series.Title.S02E01.Episode.Title which we will be able to parse and fetch metadata on. The current folder and filename structure you have is hard to parse, since even if we looked at the parent folder we would still not have the series title. Just something to consider anyway.

Please test the fix for your bug at https://www.universalmediaserver.com/uploads/UMS-2346-filename-exception-5a9ffa5.exe

**Comment by Rebel154** [comment]
@SubJunk Thank you for the quick update. I have completely uninstalled both previous versions and installed the bug-fix version from your previous post - and it now seems to be working correctly. I can see TV episodes on both my TV and the web interface.

As for my naming process, I use Sonarr and have been using it for some time without issue, with UMS, Mezzmo and Serviio. I've settled on using UMS as it allows easier profile configuration. I only recently delved into creating my own profile after buying my Panasonic 55HZ1500B UHD TV and found that UMS out of the box would not play UHD content.

You're welcome to include the configuration file in the next release. I cannot say that it is perfect, but it plays my files (HD and UHD with and without HDR) just fine with the correct audio too.

[Panasonic-HZ1500B.zip](https://github.com/UniversalMediaServer/UniversalMediaServer/files/5824233/Panasonic-HZ1500B.zip)

**Comment by mbentley** [comment]
I'm with you there @Rebel154 I ran into this as well and I have a similar situation of also using Sonarr for auto-renaming in `SxxExx.Title.ext` format.  And I do so for the same reason of the file names being too long to quickly see and parse which episode is which as my TV will only show about 15 characters max for most items in the poster style view my TV has.  Otherwise I would need to have an item selected to see the full name on my Samsung TV.

**Comment by SubJunk** [comment]
Thanks both of you for the feedback. It’s very valuable for me to be aware of the different use cases for the program.

Have you seen that in the Media Library, if we parsed the file correctly it would show up the short way too? For example for the filename “Series.Title.S01E01.Episode.Title.mkv” it could be found in “Media Library > Video > TV > Series Title > Season 1 > 01 - Episode Title”.
The benefit of that would be that because UMS is then aware of the series title, we would be able to fetch metadata that might be wanted like posters, genres, actors, etc.
